File: HappyNumber/HappyNumber.cpp
#include<unordered_set>
using namespace std;

class Solution {
public:
	bool isHappy(int n) {
		if (n < 1)
			return false;
		if (n == 1)
			return true;
		unordered_set<int> nums;
		nums.insert(n);
		while(true) {
			int tmp = 0;
			while (n != 0) {
				tmp += pow((n % 10), 2);
				n = n / 10;
			}
			if (tmp == 1)
				return true;
			else if (nums.find(tmp) != nums.end())
				return false;
			n = tmp;
			nums.insert(n);
		}
	}
};

File: HappyNumber/HappyNumber.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n == None:
            return
        count = 6
        while count:
            n_X = n
            tmp = 0
            while n_X > 1:
                i = n_X % 10
                tmp = tmp + i**2
                n_X = n_X / 10
            tmp += n_X**2
            n = tmp
            count -= 1
        if n == 1:
            return True
        else:
            return False